cogboard-app/src/main/kotlin/com/cognifide/cogboard/logStorage/LogStorage.kt

```diff
@@ -0,0 +1,293 @@
+        try {
+            val result = collection.deleteMany(lt(Log.INSERTED_ON, beforeTimestamp))
+            LOGGER.debug("Deleted ${result.deletedCount} old logs")
+        } catch (exception: MongoException) {
+            LOGGER.error("Cannot delete old logs: $exception")
+        }
+    }
+
+    /** Deletes oldest logs when logs take up too much space. */
+    private fun deleteSpaceConsumingLogs() {
+        val database = database ?: return
+        val collection = logsCollection ?: return
+
+        val size = database
+                .runCommand(Document(STATS_COMMAND, config.id))
+                .getInteger(STATS_SIZE) ?: 0
+        val maxSize = config.fileSizeMB * MB_TO_BYTES
+        if (size > 0 && size > maxSize) {
+            val deleteFactor = ((size - maxSize).toDouble() / size)
+            val logCount = collection.countDocuments()
+            val toDelete = (logCount.toDouble() * deleteFactor).toLong()
+            if (toDelete > 0) {
+                LOGGER.debug("Deleting $toDelete logs as the size $size exceeds maximum size of $maxSize")
+                deleteFirstLogs(toDelete)
+            }
+        }
+    }
+
+    /** Filters in place the logs not matching the rules. */
+    private fun filter(logs: MutableList<Log>) {
+        val regexes = enabledRegexes
+        if (regexes.isEmpty()) { return }
+        logs.retainAll { log ->
+            log.variableData.any { variable ->
+                regexes.any { it.containsMatchIn(variable.header) }
+            }
+        }
+    }
+
+    /** Deletes logs not matching to the rules from the database. */
+    fun filterExistingLogs() {
+        val collection = logsCollection ?: return
+
+        val fieldName = Log.VARIABLE_DATA + "." + LogVariableData.HEADER
+        val regexes = enabledRegexes.map { regex(fieldName, it.pattern) }
+
+        if (regexes.isEmpty()) { return }
+
+        collection.deleteMany(nor(regexes))
+    }
+
+    /** Downloads new logs and inserts the to the database. Returns the number of inserted logs. */
+    private fun downloadInsertLogs(seq: Long, skipFirstLines: Long? = null): Long {
+        var sequence = seq
+        val logs = connection
+                .getLogs(skipFirstLines)
+                .mapNotNull { parserStrategy.parseLine(it) }
+                .toMutableList()
+
+        // Filter the logs by quarantine rules
+        filter(logs)
+        logs.forEach {
+            it.seq = sequence
+            sequence += 1
+        }
+
+        logsCollection?.insertMany(logs.map { it.toDocument() })
+
+        return logs.size.toLong()
+    }
+
+    /** Checks how many logs to download, downloads them and saves them to the database. */
+    private fun downloadLogs() {
+        // Get the current settings
+        val storageConfig = collectionConfiguration
+        var lastLine = storageConfig?.lastLine ?: 0
+        var seq = storageConfig?.seq ?: 0
```

Comment:
I think those two might be `val`s

Reply:
These two variables are incremented later.